core/src/main/scala/akka/kafka/ProducerMessage.scala

```diff
@@ -35,8 +35,11 @@ object ProducerMessage {
    * `offset` of the produced message.
    */
   final case class Result[K, V, PassThrough](
-    offset: Long,
-    message: Message[K, V, PassThrough]
-  )
+      metadata: RecordMetadata,
+      //    offset: Long,
```

Comment:
or also remove the offset func for backward and let every one access it from the metadata?